hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/autoassign/AutoAssignCheckerTest.java

```diff
@@ -313,4 +316,26 @@ public void filterWithoutWeightWorksInMultiAssignmentMode() throws Exception {
         assertThat(actions).hasSize(amountOfTargets);
         assertThat(actions).allMatch(action -> !action.getWeight().isPresent());
     }
+
+    @Test
+    @Description("Verifies an auto assignment only creates actions for compatible targets")
+    void checkAutoAssignmentWithIncompatibleTargets() throws Exception {
```

Comment:
Actually only checks use case without compatibility restriction (e.g. compatible Target has no TargetType). In addition the check should include a target with a valid compatibility restriction.

Reply:
will add a valid type